### PR TITLE
Better Gallery Tagging UI & QoL fixes

### LINKED ISSUE #1420: Filtering in gallery gets confused
**Describe the bug**
I was able to confuse the gallery such that it didn't filter by tag appropriately.

**To Reproduce**
1. Navigate to gallery
2. Modify sort by & tag filtering (go back and forth between them, changing values)

**Expected behavior**
Gallery is always sorted properly by the settings chosen in the dropdown.

**Actual behavior**  
Settings chosen in the dropdown go out of sync w/ what you see in the gallery list.

**Screenshots**
Screenshot 2024-02-05 at 9 22 22 AM

### PR DESCRIPTION
**Pull request by GalaxyGamingBoy**
Resolves #1420

### CODE REVIEW

**Review by JosiasAurel — CHANGES_REQUESTED** [review]
left some comments. overall looks good to me though

File: src/pages/gallery/gallery.tsx [hunk 1]
```diff
@@ -190,27 +222,31 @@ export default function Gallery({ games, tags }: { games: GameMetadata[], tags:
...
 				{
-					gamesState.map((game: GalleryGameMetadata) => (						
-						<a
+					gamesState.map((game: GalleryGameMetadata) => (
+						<div
 							style={`display:${game.show ? "block" : "none"}`}
 							class="game"
 							href={`/gallery/${game.filename}`}
-							target="_blank"
+							onClick={() => window.open(`/gallery/${game.filename}`, '_blank')}
```
[the remark this answers is not in the document]
Comment (GalaxyGamingBoy) [reply src/pages/gallery/gallery.tsx:249]: React's `event.persist()` only works on functions, as such to be able to click the tag I had to remove the href functionality

File: src/pages/gallery/gallery.tsx [hunk 2]
```diff
@@ -43,17 +44,26 @@ export default function Gallery({ games, tags }: { games: GameMetadata[], tags:
 			});
 		setGamesState(sortGames(_games, filter.sort).map(game => ({ ...game, show: true })) as GalleryGameMetadata[]);
 
+		// set filtering text
+		const tagSelect = document.getElementById("tag-select") as HTMLSelectElement;
+		if (filter.tags.length > 1) {
+			tagSelect.selectedIndex = 2 // set filtering to multiple text
+		} else {
+			tagSelect.selectedIndex = filter.tags.length // set text accordingly
+		}
```
[the remark this answers is not in the document]
Comment (GalaxyGamingBoy) [reply src/pages/gallery/gallery.tsx:53]: Sure, the only problem is that I will have to copy paste it which generally isn't a good practice, that's why I opted for a global-ish way

File: src/pages/gallery/gallery.tsx [hunk 3]
```diff
@@ -69,7 +79,23 @@ export default function Gallery({ games, tags }: { games: GameMetadata[], tags:
 			}
 		}
 		return games;
-	}	
+	}
+
+	function countTags(games: GameMetadata[]) {
+		const tags: { [tags: string]: number } = {};
+
+		games.forEach(game => {
```
[the remark this answers is not in the document]
Comment (GalaxyGamingBoy) [reply src/pages/gallery/gallery.tsx:87]: A reduce with a map as an accumulator seems iffy style wise imo but I'll see

File: src/pages/gallery/gallery.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GalaxyGamingBoy) [reply src/pages/gallery/gallery.tsx:53]: @JosiasAurel Should I leave it be, it seems more efficient that copy-pasting it a bunch of times over the 2-3 on selects

**Comment by GalaxyGamingBoy** [comment]
@grymmy @JosiasAurel Am I free to merge?

**Pull request merged**